tests/Data/Text/Utf8/Utf8Spec.hs

```diff
@@ -33,6 +39,37 @@ spec = do
             forM_ [0..0x7f] $ flip shouldSatisfy $ \cp ->
                 mapCp Char.toLower cp == Utf8.toLowerAscii cp
 
+    describe "slicing functions" $ do
+
+        it "satisfies the example in Data.Text.Utf8" $ do
+            let begin = Utf8.CodeUnitIndex 2
+            let length_   = Utf8.CodeUnitIndex 6
+            Utf8.unsafeSliceUtf8 begin length_ slicingExample `shouldBe` Utf8.pack "DEFGHI"
+            Utf8.unsafeCutUtf8 begin length_ slicingExample `shouldBe` (Utf8.pack "BC", Utf8.pack "JKL")
+
+        prop "unsafeSliceUtf8 and unsafeCutUtf8 are complementary" $ \ (SlicingExampleIndices begin length_ :: SlicingExampleIndices) -> do
+            let (prefix, suffix) = Utf8.unsafeCutUtf8 begin length_ slicingExample
+            Utf8.concat [prefix, Utf8.unsafeSliceUtf8 begin length_ slicingExample, suffix] `shouldBe` slicingExample
+
+
+-- | Example shown in section "Slicing Functions" in 'Data.Text.Utf8".
+slicingExample :: Utf8.Text
+slicingExample = Utf8.Text (stringToByteArray "ABCDEFGHIJKLMN") 1 11
+
+data SlicingExampleIndices = SlicingExampleIndices Utf8.CodeUnitIndex Utf8.CodeUnitIndex
+    deriving Show
+
+instance Arbitrary SlicingExampleIndices where
+    arbitrary = do
+        let exampleLength = unCodeUnitIndex $ Utf8.lengthUtf8 slicingExample
+
+        begin <- choose (0, exampleLength)
+        length_ <- choose (0, exampleLength - begin)
+
+        pure $ SlicingExampleIndices (Utf8.CodeUnitIndex begin) (Utf8.CodeUnitIndex length_)
+
+        where unCodeUnitIndex (Utf8.CodeUnitIndex i) = i
```

Comment:
Quite a bit of boilerplate for a simple test but it's pretty in a way :lady_beetle: